server/auth/types/openid/routes.ts

```diff
@@ -54,12 +54,17 @@ export class OpenIdAuthRoutes {
       {
         path: `/auth/openid/login`,
         validate: {
-          query: schema.object({
-            code: schema.maybe(schema.string()),
-            nextUrl: schema.maybe(schema.string()),
-            state: schema.maybe(schema.string()),
-            refresh: schema.maybe(schema.string()),
-          }),
+          query: schema.object(
+            {
+              code: schema.maybe(schema.string()),
+              nextUrl: schema.maybe(schema.string()),
+              state: schema.maybe(schema.string()),
+              refresh: schema.maybe(schema.string()),
+            },
+            {
+              unknowns: 'allow',
+            }
+          ),
```

Comment:
Would an explicit allow list be better? What is the risk if a malicious user adds additional query parameters?

Reply:
the additional parameters will anyways not being used in the API handler, Given different IDPs may have different OIDC implementations, having an allow list may still expose errors like `opendistro-for-elasticsearch/opendistro-build#416` for other parameters. Let me know if you came up with any risk with allowing all params.